models/tags.js

```diff
@@ -0,0 +1,88 @@
+const firestore = require("../utils/firestore");
+const tagModel = firestore.collection("tags");
+
+/**
+ *
+ * @param tagData { Object }: Data of the tag
+ * @returns {Promise<tagModel|Object>}
+ */
+
+const addTag = async (tagData) => {
+  try {
+    tagData.type = tagData.type.toUpperCase();
```

Comment:
check wether there is already a tag with the same name and type present in the database